src/api/initialDataTypes.js

```diff
@@ -120,6 +120,7 @@ export type RawInitialDataRealmUser = {|
   is_admin: boolean,
   realm_users: Array<{| ...User, avatar_url?: string | null |}>,
   realm_non_active_users: Array<{| ...User, avatar_url?: string | null |}>,
+  realm_mandatory_topics: boolean,
```

Comment:
We already have this on `InitialDataRealm`, so no need to add it elsewhere.

The right place for it is indeed `InitialDataRealm`. The organization of these different types is described at the place where they're all combined together:
```js
// Initial data snapshot sent in response to a `/register` request,
// after validation and transformation.
export type InitialData = {|
  // The server sends different subsets of the full available data,
  // depending on what event types the client subscribes to with the
  // `fetch_event_types` field of the `/register` request. We name these
  // subsets after the event types that cause them to be included.
```

This field is controlled by the `realm` event type being present; from the docs at https://zulip.com/api/register-queue :
> `realm_mandatory_topics`: boolean Present if `realm` is present in `fetch_event_types`.

so `InitialDataRealm` is the right place.

It'd belong on `InitialDataRealmUser` (and therefore here on `RawInitialDataRealmUser`, which that's based on) if it were instead controlled by `realm_user`.

(This organization isn't totally obvious, even though there is that one comment about it. I may go and add some short jsdoc on the individual types here.)